src/dmd/expressionsem.d

```diff
@@ -3587,36 +3587,58 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
         else if (tb.isscalar())
         {
+            Identifier id;
+            if (tb.isZeroInit(exp.loc) == true)
```

Comment:
Can't the implementation of `___d_newitemT` figure this out so there's only one function that's needed?

Reply:
Moved this into `___d_newitemT`'s implementation